org-roam-capture.el

```diff
@@ -414,27 +435,58 @@ aborted, we do the following:
+                                  (file-name-as-directory)
+                                  (concat org-roam-dailies-capture--file-name-default))
+                                (user-error "`org-roam-dailies-directory' cannot be nil")))
+                           ('ref
+                            org-roam-capture-ref--file-name-default)
+                           (_
+                            org-roam-capture--file-name-default))))
          (new-id (s-trim (org-roam-capture--fill-template
                           name-templ)))
          (file-path (org-roam--file-path-from-id new-id))
          (roam-head (or (org-roam-capture--get :head)
-                        org-roam-capture--header-default))
+                        (pcase org-roam-capture--context
+                          ('dailies
+                           org-roam-dailies-capture--header-default)
+                          ('ref
+                           org-roam-capture-ref--header-default)
+                          (_
+                           org-roam-capture--header-default))))
          (org-template (org-capture-get :template))
          (roam-template (concat roam-head org-template)))
-    (unless (file-exists-p file-path)
+    (unless (or (file-exists-p file-path)
+                (cl-some (lambda (buffer)
+                           (string= (buffer-file-name buffer)
+                                    file-path))
+                         (buffer-list)))
       (make-directory (file-name-directory file-path) t)
       (org-roam-capture--put :orig-no-save (org-capture-get :no-save)
                              :new-file t)
-      (org-capture-put :template
-                       ;; Fixes org-capture-place-plain-text throwing 'invalid search bound'
-                       ;; when both :unnarowed t and "%?" is missing from the template string;
-                       ;; may become unnecessary when the upstream bug is fixed
-                       (if (s-contains-p "%?" roam-template)
-                           roam-template
-                         (concat roam-template "%?"))
-                       :type 'plain
-                       :no-save t))
+      (pcase org-roam-capture--context
+        ('dailies
+         ;; Populate the header of the daily file before capture to prevent it
+         ;; from appearing in the buffer-restriction
+         (save-window-excursion
+           (find-file file-path)
+           (insert (substring (org-capture-fill-template (concat roam-head "*"))
+                              0 -2))
+           (set-buffer-modified-p nil))
+         (org-capture-put :template org-template))
+        (_
+         (org-capture-put :template
+           ;; Fixes org-capture-place-plain-text throwing 'invalid search bound'
+           ;; when both :unnarowed t and "%?" is missing from the template string;
+           ;; may become unnecessary when the upstream bug is fixed
```

Comment:
This was fixed as of: https://code.orgmode.org/bzg/org-mode/commit/09086b7e75a4ccb27f141d2cd495c253f1b6b025

Reply:
Splendid: I'll amend the code.

Reply:
Addressed in 7325b0af7fd053be506dfb32762c8fb8730ad390.